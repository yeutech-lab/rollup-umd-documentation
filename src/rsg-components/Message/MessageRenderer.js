import React from 'react';
import PropTypes from 'prop-types';
import Markdown from 'rsg-components/Markdown';
import Styled from 'rsg-components/Styled';

const styles = ({ space }) => ({
	root: {
		marginBottom: space[4],
	},
});

export function MessageRenderer({ classes, children }) {
	return (
		<div className={classes.root}>
			<Markdown text={Array.isArray(children) ? children.join('\n') : children} />
		</div>
	);
}

MessageRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node.isRequired,
};

export default Styled(styles)(MessageRenderer);
