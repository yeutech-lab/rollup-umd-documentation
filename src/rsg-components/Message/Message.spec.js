import React from 'react';
import { MessageRenderer } from './MessageRenderer';

it('renderer should render message', () => {
	const message = 'Hello *world*!';
	const actual = shallow(<MessageRenderer classes={{}}>{message}</MessageRenderer>);

	expect(actual).toMatchSnapshot();
});

it('renderer should render message for array', () => {
	const messages = ['Hello *world*!', 'Foo _bar_'];
	const actual = shallow(<MessageRenderer classes={{}}>{messages}</MessageRenderer>);

	expect(actual).toMatchSnapshot();
});
