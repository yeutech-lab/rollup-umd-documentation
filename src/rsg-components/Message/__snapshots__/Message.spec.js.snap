// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render message 1`] = `
<div>
  <Markdown
    text="Hello *world*!"
  />
</div>
`;

exports[`renderer should render message for array 1`] = `
<div>
  <Markdown
    text="Hello *world*!
Foo _bar_"
  />
</div>
`;
